### PR TITLE
20991 int - Add Counseling Office Column and Filtering to TIO Queue

### PR DESCRIPTION
**Pull request by JonSpight**
## Summary
When a customer selects an origin duty location assigned to government counseling, then they will also be asked to select a counseling office within the same GBLOC. This field will be called "Counseling Office".

How to test
1. Login as a customer and create a 2 both with different counseling offices.
2. Login as a Services Counselor and submit the new move.
3 Login as TOO, approve both moves.
4. Crete multiple payment requests for each move
5. Login as TIO and filter the Counseling Office Task Order Queue for a move to verify filtering works.
6. Login as TIO and sort the Counseling Office Task Order Queue for a move to verify sort works.

## Screenshots
[B-20991-508Screenshots.pdf](https://github.com/user-attachments/files/17740713/B-20991-508Screenshots.pdf)

### CODE REVIEW

**Review by loganwc — APPROVED** [review]

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go [hunk 1]
```diff
@@ -27,6 +28,85 @@ func TestMove(t *testing.T) {
...
+	paymentRequests[0].MoveTaskOrder.TIOAssignedUser = &officeUser
+	paymentRequests[0].MoveTaskOrder.CounselingOffice = &transportationOffice
+
+	var officeUsers models.OfficeUsers
+	officeUsers = append(officeUsers, officeUser)
+
+	suite.Run("Test Counseling Offices and TIO AssignedUser ", func() {
+		result := QueuePaymentRequests(&paymentRequests, officeUsers, officeUser, true, false)
+		suite.NotNil(result)
+	})
```
Comment (danieljordan-caci) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go:69]: Can you expand more on this test? Just seems that it's testing that it's not nil, but the test description is telling me it should be checking for the counseling office and assigned user values not being nil.

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go [hunk 2]
```diff
@@ -27,6 +28,85 @@ func TestMove(t *testing.T) {
...
+
+	paymentRequests := models.PaymentRequests{pr2}
+
+	var officeUsers models.OfficeUsers
+	officeUsers = append(officeUsers, officeUser)
+
+	suite.Run("attach available office users if move is assignable", func() {
+		result := QueuePaymentRequests(&paymentRequests, officeUsers, officeUser, false, false)
+		suite.NotNil(result)
+	})
```
Comment (danieljordan-caci) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go:107]: Same thing here - your test description and the actual test contradict each other. Just adding a little specificity would be nice. Thanks! Nice start!

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go [hunk 3]
```diff
@@ -27,6 +28,93 @@ func TestMove(t *testing.T) {
...
+				ProvidesCloseout: true,
+			},
+		},
+	}, nil)
+	var officeUsers models.OfficeUsers
+	officeUsers = append(officeUsers, officeUser)
+	var paymentRequestsQueue = QueuePaymentRequests(&paymentRequests, officeUsers, officeUser, false, false)
+
+	suite.Run("Test Payment request is assignable due to not being assigend", func() {
+		// paymentRequests := QueuePaymentRequests(&paymentRequests, officeUsers, officeUser, false, false)
```
Comment (danieljordan-caci) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go:73]: left some commented out code in here

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go [hunk 4]
```diff
@@ -27,6 +28,93 @@ func TestMove(t *testing.T) {
...
+		paymentRequestCopy := *paymentRequestsQueue
+		suite.NotNil(paymentRequestsQueue)
+		suite.IsType(paymentRequestsQueue, &ghcmessages.QueuePaymentRequests{})
+		suite.Nil(paymentRequestCopy[0].AssignedTo)
+	})
+
+	suite.Run("Test Payment request has no counseling office", func() {
+		// paymentRequests := QueuePaymentRequests(&paymentRequests, officeUsers, officeUser, false, false)
```
Comment (danieljordan-caci) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go:81]: left some commented out code in here

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go [hunk 5]
```diff
@@ -27,6 +28,93 @@ func TestMove(t *testing.T) {
...
+		suite.Nil(paymentRequestCopy[0].CounselingOffice)
+	})
+
+	paymentRequests[0].MoveTaskOrder.TIOAssignedUser = &officeUserTIO
+	paymentRequests[0].MoveTaskOrder.CounselingOffice = &transportationOffice
+
+	paymentRequestsQueue = QueuePaymentRequests(&paymentRequests, officeUsers, officeUser, false, false)
+
+	suite.Run("Test PaymentRequest has both Counseling Office and TIO AssignedUser ", func() {
+		// PaymentRequests := QueuePaymentRequests(&paymentRequests, officeUsers, officeUser, true, false)
```
Comment (danieljordan-caci) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload_test.go:94]: left some commented out code in here

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Tests look good - just a little cleanup and you're G2G.

**Review by danieljordan-caci — APPROVED** [review]
